.github/workflows/ci.yml

```diff
@@ -8,60 +8,93 @@ on:
       - '**'
   pull_request: {}
 
+env:
+  COLUMNS: 150
+
 jobs:
   lint:
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ['3.12']
 
     steps:
       - uses: actions/checkout@v4
 
-      - name: set up python
-        uses: actions/setup-python@v5
+      - uses: astral-sh/setup-uv@v3
         with:
-          python-version: '3.10'
+          enable-cache: true
 
-      - name: install
-        run: |
-          pip install -r requirements/pyproject.txt && pip install -r requirements/linting.txt
+      - name: Install dependencies
+        run: uv sync --python ${{ matrix.python-version }} --group lint --all-extras
 
-      - uses: pre-commit/action@v3.0.1
+      - uses: pre-commit/action@v3.0.0
         with:
           extra_args: --all-files --verbose
+        env:
+          SKIP: no-commit-to-branch
 
   test:
-    name: test py${{ matrix.python-version }} on ${{ matrix.os }}
+    name: test py${{ matrix.python-version }}
+    runs-on: ubuntu-latest
 
     strategy:
       fail-fast: false
       matrix:
-        os: [ubuntu, macos, windows]
-        python-version: ['3.8', '3.9', '3.10', '3.11', '3.12']
+        python-version: ['3.9', '3.10', '3.11', '3.12', '3.13']
+
+    env:
+      UV_PYTHON: ${{ matrix.python-version }}
 
-    runs-on: ${{ matrix.os }}-latest
     steps:
       - uses: actions/checkout@v4
 
-      - name: set up python
-        uses: actions/setup-python@v5
+      - uses: astral-sh/setup-uv@v3
         with:
-          python-version: ${{ matrix.python-version }}
+          enable-cache: true
 
-      - run: |
-          pip install -r requirements/pyproject.txt && pip install -r requirements/testing.txt
+      - name: Install extra dependencies
+        run: uv sync --group dev --all-extras
```

Comment:
I believe `dev` is installed by default.

Reply:
If I removed this it fail because seems it's not finding the extra packages needed to run tests https://github.com/pydantic/pydantic/blob/15f693a25656b3b004d537d2b6371f81336c4cc1/.github/workflows/ci.yml#L129

i will do it like this